tests/test_browser.py

```diff
@@ -38,6 +38,48 @@ def test_submit_online(httpbin):
     assert 'MechanicalSoup' in json["headers"]["User-Agent"]
 
 
+def test_get_request_kwargs(httpbin):
+    """Return kwargs without a submit"""
+    browser = mechanicalsoup.Browser()
+    page = browser.get(httpbin + "/forms/post")
+    form = page.soup.form
+    form.find("input", {"name": "custname"})["value"] = "Philip J. Fry"
+    request_kwargs = browser.get_request_kwargs(form, page.url)
+    assert "method" in request_kwargs
+    assert "url" in request_kwargs
+    assert "data" in request_kwargs
+    assert ("custname", "Philip J. Fry") in request_kwargs["data"]
+
+
+def test_get_request_kwargs_when_method_is_in_kwargs(httpbin):
+    """Raise TypeError exception"""
+    browser = mechanicalsoup.Browser()
+    page = browser.get(httpbin + "/forms/post")
+    form = page.soup.form
+    kwargs = {"method": "post"}
+    type_error_raised = None
+    try:
+        browser.get_request_kwargs(form, page.url, **kwargs)
+    except TypeError:
+        type_error_raised = True
```

Comment:
Please use [pytest.raises](https://docs.pytest.org/en/stable/reference.html?highlight=raises#pytest.raises) to assert that an exception is raised.

Reply:
Thanks, I'll rewrite it using pytest.raises.